### ISSUE TITLE
Recording loss metrics

### ISSUE THREAD

**Issue by nicholastu2** [ultralytics/ultralytics#8021]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello, I want to be able to save a training run in a folder with a certain name that includes info about the run. For example, if I trained YOLOv8 with 8 epochs and the lowest cls_loss out of those epochs was 5.719, then I want to save that run in a folder called ep8cls_loss5.719. 

I know you can choose what the name of the folder is with the 'name' parameter for YOLO.train, but I could only figure out how to save the folder as how many epochs we chose, but I want to also be able to include about the loss metrics.

Does anyone know how I could do this? Is the information about any of the loss metrics contained in the train object after an epoch is finished?

**Comment by glenn-jocher** [comment]
@nicholastu2 hello! To achieve custom folder naming based on loss metrics, you'll need to implement a custom callback or modify the training script. The loss metrics are logged after each epoch and can be accessed within the training loop. You could capture the desired loss value and then, at the end of training, rename the output directory accordingly.

Keep in mind that this would require a bit of coding on your side, as the current functionality of YOLOv8 does not support dynamic naming of output directories based on loss metrics out-of-the-box. You can refer to the training loop in the source code and the documentation for guidance on accessing loss values.

Remember to keep track of the best loss value during training and use it to rename the folder after the training process completes. Happy coding! 😊
